File: Frontend/js/bar.js
// Fonction pour récupérer les détails de l'annonce depuis l'API
function fetchAnnonceDetails() {
    const urlParams = new URLSearchParams(window.location.search);
    const annonceId = urlParams.get('id');
    const xhr = new XMLHttpRequest();
    xhr.open('POST', 'http://localhost:3000/scrapping/getOne', true);
    xhr.setRequestHeader('Content-Type', 'application/json');
    xhr.send(JSON.stringify({
        "id": annonceId
    }));
    xhr.onreadystatechange = function () {
        let bar = JSON.parse(xhr.responseText);
        console.log(bar);
        document.getElementById('barName').innerHTML = bar.name;
        document.getElementById('barAdresse').innerHTML = bar.adresse;
        document.getElementById('barImage').src = bar.image;
        document.getElementById('barDescription').innerHTML = bar.description;
        document.getElementById('barTelephone').innerHTML = bar.telephone;
        document.getElementById('barHoraire').innerHTML = bar.horaire;
    };
}

fetchAnnonceDetails();

File: Frontend/proxibar/lib/bar.dart
class Bar {
  final int id;
  final String name;
  final String adresse;
  final String horaire;
  final String telephone;
  final String image;
  final String latitude;
  final String longitude;
  final String carte;
  final String distance;

  Bar(
      {required this.id,
      required this.name,
      required this.adresse,
      required this.horaire,
      required this.telephone,
      required this.image,
      required this.latitude,
      required this.longitude,
      required this.carte,
      required this.distance});

  factory Bar.fromJson(Map<dynamic, dynamic> json) {
    return Bar(
        id: json['bar']['id'],
        name: json['bar']['name'],
        adresse: json['bar']['adresse'],
        horaire: json['bar']['horaire'],
        telephone: json['bar']['telephone'],
        image: json['bar']['image'],
        latitude: json['bar']['latitude'],
        longitude: json['bar']['longitude'],
        carte: json['bar']['carte'],
        distance: json['distance']);
  }
}
